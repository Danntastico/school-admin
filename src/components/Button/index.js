import React from "react";
import { Btn } from "./styles";
import PropTypes from "prop-types";

export const Button = ({ onClick, label = "Button", isDisabled }) => {
  return (
    <Btn onClick={onClick} disabled={isDisabled}>
      {label}
    </Btn>
  );
};
