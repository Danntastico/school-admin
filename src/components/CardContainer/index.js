import React from 'react';
import { Container } from './styles';
export const CardContainer = ({ children }) => {
  return <Container>{children}</Container>;
};
