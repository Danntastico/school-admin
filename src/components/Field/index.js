import React from "react";
import { Input, Label, FieldContainer } from "./styles";
export const Field = ({
  disabled = false,
  fieldType,
  handleInputChange,
  inputType,
  label,
  name,
  placeholder,
  value,
  children,
}) => {
  const fieldRender = () => {
    switch (fieldType) {
      case "textarea":
        return (
          <textarea
            autoComplete="off"
            disabled={disabled}
            name={name}
            id={name}
            onChange={handleInputChange}
            value={value}
          />
        );
      case "select":
        return (
          <select name={name} value={value} onChange={handleInputChange}>
            {children}
          </select>
        );
      default:
        return (
          <Input
            autoComplete="off"
            disabled={disabled}
            id={name}
            name={name}
            onChange={handleInputChange}
            placeholder={placeholder}
            type={inputType}
            value={value}
            required={true}
          />
        );
    }
  };
  return (
    <FieldContainer>
      <Label htmlFor={name}> {label}</Label>
      {fieldRender()}
    </FieldContainer>
  );
};
