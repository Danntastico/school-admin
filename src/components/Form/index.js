import React from 'react';
import { FormHeader, FormTitle, FormBody, FormFooter } from './styles';
export const Form = ({ children, title, onSubmit }) => {
  return (
    <form onSubmit={onSubmit}>
      <FormHeader>
        <FormTitle> {title} </FormTitle>
      </FormHeader>
      <FormBody>{children[0]}</FormBody>
      <FormFooter>{children[1]}</FormFooter>
    </form>
  );
};
