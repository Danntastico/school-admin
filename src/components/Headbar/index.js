import React from 'react';

export const Headbar = ({ title }) => {
  return (
    <header>
      <h3>title</h3>
    </header>
  );
};
