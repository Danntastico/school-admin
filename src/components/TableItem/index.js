import React from 'react';
import { MdAccountCircle, MdLibraryBooks } from 'react-icons/md';

import { ItemContainer } from './styles';

const Icon = (type) => {
  switch (type) {
    case 'subject':
      return <MdLibraryBooks />;
    default:
      return <MdAccountCircle />;
  }
};

const testConfig = {
  id: '12345',
  lastName: 'lastName',
  name: 'name',
};

export const TableItem = ({ config = testConfig, type = 'teacher' }) => {
  const icon = Icon(type);
  const labels = Object.values(config);

  return (
    <ItemContainer>
      {icon}
      {labels.map((label) => (
        <p key={label}>{label}</p>
      ))}
      <button>Toggle Button</button>
    </ItemContainer>
  );
};
