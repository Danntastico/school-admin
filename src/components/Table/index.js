import React from 'react';
import { TableContainer, TableHeader, TableContent } from './styles';
import { Button } from '../Button';

export const Table = ({ title = 'Items', children }) => {
  return (
    <TableContainer>
      <TableHeader>
        <h2>{title}</h2>
        <Button>Add New Item</Button>
      </TableHeader>
      <TableContent>{children}</TableContent>
    </TableContainer>
  );
};
