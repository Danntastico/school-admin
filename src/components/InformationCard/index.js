import React from "react";
import { MdEdit } from "react-icons/md";
import { InfoCard, InfoHead, InfoBody, InfoTitle } from "./styles";
import { EditBtn } from "../Form/styles";
export const InformationCard = ({
  children,
  handleClickOnEdit,
  title,
  isEditable,
}) => {
  return (
    <InfoCard>
      <InfoHead>
        <InfoTitle>{title}</InfoTitle>
        {isEditable && (
          <EditBtn onClick={handleClickOnEdit}>
            <MdEdit />
          </EditBtn>
        )}
      </InfoHead>
      <InfoBody>{children}</InfoBody>
    </InfoCard>
  );
};
