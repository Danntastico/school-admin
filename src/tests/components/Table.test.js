import React from 'react';
import { shallow, mount } from 'enzyme';
import { Table } from '../../components/Table';
import { TableItem } from '../../components/TableItem';

describe('Unit test for <Table>', () => {
  const testConfig = {
    id: '12345',
    lastName: 'lastName',
    name: 'name',
  };
  const wrapper = mount(
    <Table title='test Items'>
      <TableItem config={testConfig} type='teacher' />
    </Table>
  );

  test('Should show component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render Title correctly', () => {
    const title = wrapper.find('h2');
    expect(title.text().trim()).toBe('test Items');
  });
});
