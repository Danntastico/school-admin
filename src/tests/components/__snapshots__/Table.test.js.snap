// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for <Table> Should show component correctly 1`] = `
<div
  className="sc-AxjAm iPUaXY"
>
  <div
    className="sc-AxirZ gGFIeF"
  >
    <h2>
      test Items
    </h2>
    <button
      className="sc-AxhCb lkXtCa"
    >
      Button
    </button>
  </div>
  <div
    className="sc-AxiKw fcEhuF"
  >
    <div
      className="sc-AxhUy flixkA"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          key="0"
        />
      </svg>
      <p
        key="12345"
      >
        12345
      </p>
      <p
        key="lastName"
      >
        lastName
      </p>
      <p
        key="name"
      >
        name
      </p>
      <button>
        Toggle Button
      </button>
    </div>
  </div>
</div>
`;
